Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -0,0 +1,194 @@
+    if (!success) {
+        return success;
+    }
+    success = [self microphoneInput:YES];
+    if (!success) {
+        return success;
+    }
+
+    [self setUpStreamFormat];
+
+    OSStatus status = AUGraphInitialize(processingGraph);
+    CheckStatus(status, @"Failed to initialize processing graph");
+    if (status != noErr) {
+        return NO;
+    }
+
+    AUGraphStart(processingGraph);
+    CheckStatus(status, @"Failed to start processing graph");
+
+    return (status == noErr);
+}
+
+-(BOOL)stopAudioFlow:(NSError **)error
+{
+    CheckStatus(AUGraphStop(processingGraph), @"Failed to stop processing graph");
+    CheckStatus(AUGraphUninitialize(processingGraph), @"Unable to uninitialize graph");
+    AVAudioSession *session = [AVAudioSession sharedInstance];
+
+    bool success = [session setActive:NO error:error];
+
+    return success;
+}
+
+-(BOOL)microphoneInput:(BOOL)enable;
+{
+    UInt32 enableInput = (enable)? 1 : 0;
+    OSStatus status = AudioUnitSetProperty(
+                                           ioUnit,
+                                           kAudioOutputUnitProperty_EnableIO,//property we are changing
+                                           kAudioUnitScope_Input,
+                                           kInputBus,
+                                           &enableInput,
+                                           sizeof (enableInput)
+                                           );
+    CheckStatus(status, @"Unable to enable/disable input");
+    return (status == noErr);
+}
+
+-(BOOL)outputEnable:(BOOL)enable error:(NSError **)error
+{
+    return YES;
+}
+
+#pragma mark - Audio Status
+-(BOOL)isAudioRunning
+{
+    Boolean running;
+    CheckStatus(AUGraphIsRunning(processingGraph,
+                                 &running),
+                @"Unable to determine if AUGraph is running");
+    return running;
+}
+
+#pragma mark - Private
+-(BOOL)startAudioSession:(NSError **)error
+{
+    AVAudioSession *session = [AVAudioSession sharedInstance];
+    bool success = [session setCategory:AVAudioSessionCategoryPlayAndRecord
+                   error:error];
+    if (!success) {
+        return NO;
+    }
+    success = [session setActive:YES error:error];
+    if (!success) {
+        return NO;
+    }
+
+    self.sampleRate = session.sampleRate;
+
+    return YES;
+}
+
+-(void)setUpStreamFormat
+{
+    //Always initialize the fields of a new audio stream basic description structure to zero
+    UInt32 bytesPerSample = sizeof (SInt32);
+
+    AudioStreamBasicDescription asbd = {0};
+    asbd.mSampleRate = self.sampleRate;
+    asbd.mFormatID = kAudioFormatLinearPCM;
+    asbd.mFormatFlags = kAudioFormatFlagIsFloat | kAudioFormatFlagsNativeEndian | kAudioFormatFlagIsPacked | kAudioFormatFlagIsNonInterleaved;
+    asbd.mChannelsPerFrame = 2;
+    asbd.mBytesPerFrame = bytesPerSample;
+    asbd.mBitsPerChannel = 8 * bytesPerSample;
+    asbd.mFramesPerPacket = 1;
+    asbd.mBytesPerPacket = bytesPerSample;
```

Comment:
Are you sure about all this configuration? E.g. have you checked that toxav uses native endian, and so on?